### ISSUE TITLE
Predictive Analytics For Improving Developer Experience

### ISSUE THREAD

**Issue by moorara** [renovatebot/renovate#4218]
**What would you like Renovate to be able to do?**

I would like to see Renovate will give us some analytics and **predictive analytics**, so organizations can improve and optimize their experience. Here are some of the challenges:

Some packages are released every day and we need to reduce the frequency of Renovate PR for them.

Some packages are dependent on each other, so a new version of a given package will also come with releases of other packages. In this case we need to group them.

It would be nice if Renovate can give some suggestion about the packages that can be grouped or the packages that need to be update weekly, through GitHub issues and pull requests.

**Describe the solution you'd like**

Renovate will open issue or pull request and suggests updating Renovate configurations to improve and optimize the DX.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by rarkins** [comment]
For packages that release every day, wouldn’t scheduling work best for that? (eg only update on weekends, or update monthly). 

For packages that may release in bursts but then have quiet periods, we’re working on a feature to introduce a “stability” concept (eg x days without a new patch)

Renovate already does a lot of automatic grouping, and the grouping is community-contributed too. Make sure you’re using our config:base preset or at least the grouping sub-presets it extends. For things we miss (packages that should always be grouped together) issues or PRs are welcome to https://github.com/renovatebot/presets

**Comment by moorara** [comment]
@rarkins Sorry, for the confusion. I am aware of features like _grouping_ and _presets_ that already have some of these configurations.

I'm more interested in seeing **Renovate suggests configurations for grouping packages and custom schedules**.

With private packages and dependencies between them that each organization have, figuring out which packages should be grouped and which should have weekly or monthly updates is a manual process and I think Renovate can automated this discovery process.

**Comment by rarkins** [comment]
I agree with you on the idea, it can be a bit challenging to implement though. Something I found quickly with Renovate Bot is that people have very different opinions on things, and also it's really easy to accidentally "annoy" people too. The solution has mostly been:
- Keep the default settings fairly unopinionated
- Make things opt-in wherever possible, because it's better to have 20% "miss out" on a setting they don't know about than to have 10% annoyed by it being "forced" on them

I guess the core question is: assuming it can develop suggestions, how can Renovate "suggest" things to users in an unobtrusive way?
- Pull Requests (that's pretty pushy)
- Perhaps within an issue (a little hard to get the UX right, e.g. how to dismiss suggestions, etc)
- Via a dashboard/portal?

**Comment by HonkingGoose** [comment]
@rarkins did you intend to add this to **Done** on your **Renovate** board? 

If this is marked done on your board, shouldn't this issue be closed as well?

I think maybe you made a mistake? I have not seen the bot offer configurations to me for grouping packages/custom schedules yet...

**Comment by rarkins** [comment]
I'm not sure what exactly happened, but some issues are marked as Done incorrectly.
